### PR TITLE
Add Read,Update,Delete,List repo functions.

### PR DESCRIPTION
**Pull request by talanknight**
@jimlambrt please pay special attention to the oplog changes.

### CODE REVIEW

File: internal/auth/oidc/account.go [hunk 1]
```diff
@@ -118,14 +118,14 @@ func (a *Account) SetTableName(n string) {
 }
 
 // oplog will create oplog metadata for the Account.
-func (c *Account) oplog(op oplog.OpType, authMethodScopeId string) oplog.Metadata {
+func (c *Account) oplog(op oplog.OpType) oplog.Metadata {
 	metadata := oplog.Metadata{
-		"resource-public-id": []string{c.AuthMethodId}, // the auth method is the root aggregate
-		"resource-type":      []string{"oidc auth account"},
+		"resource-public-id": []string{c.GetPublicId()},
+		"resource-type":      []string{"password account"},
```
Comment (jimlambrt) [inline internal/auth/oidc/account.go:124]: looks like cut/paste oversight.  this isn't a password account.

File: internal/auth/oidc/account.go [hunk 2]
```diff
@@ -118,14 +118,14 @@ func (a *Account) SetTableName(n string) {
 }
 
 // oplog will create oplog metadata for the Account.
-func (c *Account) oplog(op oplog.OpType, authMethodScopeId string) oplog.Metadata {
+func (c *Account) oplog(op oplog.OpType) oplog.Metadata {
```
Comment (jimlambrt) [inline internal/auth/oidc/account.go:121]: why have you removed scope id?  This is an issue that we're going to have to fix across all the resources that don't have it.  Please add scope id back into the metadata.

File: internal/auth/oidc/repository_auth_method.go [hunk 3]
```diff
@@ -196,7 +196,7 @@ func upsertOplog(ctx context.Context, w db.Writer, oplogWrapper wrapping.Wrapper
 	if err != nil {
 		return errors.Wrap(err, op, errors.WithMsg("unable to get ticket"))
 	}
-	metadata := acct.oplog(operation, scopeId)
+	metadata := acct.oplog(operation)
```
Comment (jimlambrt) [inline internal/auth/oidc/repository_auth_method.go:199]: please add scope id back into the metadata. (see my earlier comment)

File: internal/auth/oidc/testing.go [hunk 4]
```diff
@@ -146,6 +146,7 @@ func TestSortAuthMethods(t *testing.T, methods []*AuthMethod) {
 
 // TestAccount creates a test oidc auth account.
 func TestAccount(t *testing.T, conn *gorm.DB, am *AuthMethod, issuerId *url.URL, subjectId string, opt ...Option) *Account {
+	t.Helper()
```
Comment (jimlambrt) [inline internal/auth/oidc/testing.go:149]: ty!

File: internal/auth/oidc/repository_account_test.go [hunk 5]
```diff
@@ -0,0 +1,800 @@
...
+				assert.Equal(tt.wantErrMsg, err.Error())
+				return
+			}
+			require.NoError(err)
+			assert.EqualValues(tt.want, got)
+		})
+	}
+}
+
+func TestRepository_DeleteAccount(t *testing.T) {
```
Comment (jimlambrt) [inline internal/auth/oidc/repository_account_test.go:89]: consider adding a test for missing scope id.

File: internal/auth/oidc/repository_account_test.go [hunk 6]
```diff
@@ -0,0 +1,800 @@
...
+			assert.NoError(err)
+			require.NotNil(repo)
+			got, err := repo.ListAccounts(context.Background(), am.GetPublicId(), tt.listOpts...)
+			require.NoError(err)
+			assert.Len(got, tt.wantLen)
+		})
+	}
+}
+
+func TestRepository_UpdateAccount(t *testing.T) {
```
Comment (jimlambrt) [inline internal/auth/oidc/repository_account_test.go:333]: perhaps add tests for missing version and scope id?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
Looks great! just a few suggestions.

File: internal/auth/oidc/repository_account_test.go (hunk 6 above)
Comment (talanknight) [reply internal/auth/oidc/repository_account_test.go:333]: Added. Thanks!

File: internal/auth/oidc/repository_account_test.go (hunk 5 above)
Comment (talanknight) [reply internal/auth/oidc/repository_account_test.go:89]: Added. Thanks!

File: internal/auth/oidc/repository_auth_method.go (hunk 3 above)
Comment (talanknight) [reply internal/auth/oidc/repository_auth_method.go]: Added. Thanks!

**Review by jimlambrt — APPROVED** [review]
Looks great!

**Pull request merged**